test/unit/builtins/array.spec.ts

```diff
@@ -497,6 +498,7 @@ test.each([
 });
 
 test.each([
+    { array: [[]], map: <T>(v: T) => v, expected: [] },
```

Comment:
Could you also add testcase `{ array: [1, 2, 3], map: (v: number) => ({ a: v * 2 }), expected: [{ a: 2 }, { a: 4 }, { a: 6 }] },` to this list (passes on current master)